setup/common/040_ValidateSignCert.rb

```diff
@@ -1,45 +1,51 @@
 test_name "Validate Sign Cert" do
-  skip_test 'not testing with puppetserver' unless @options['is_puppetserver']
-  hostname = on(master, 'facter hostname').stdout.strip
-  fqdn = on(master, 'facter fqdn').stdout.strip
-  puppet_version = on(master, puppet("--version")).stdout.chomp
-
-  if master.use_service_scripts?
-    step "Ensure puppet is stopped"
-    # Passenger, in particular, must be shutdown for the cert setup steps to work,
-    # but any running puppet master will interfere with webrick starting up and
-    # potentially ignore the puppet.conf changes.
-    on(master, puppet('resource', 'service', master['puppetservice'], "ensure=stopped"))
-  end
-
-  step "Clear SSL on all hosts"
+  need_to_run = false
   hosts.each do |host|
-    ssldir = on(host, puppet('agent --configprint ssldir')).stdout.chomp
-    # preserve permissions for master's ssldir so puppetserver can read it
-    on(host, "rm -rf '#{ssldir}/*'")
+    need_to_run ||= !host['use_existing_container']
   end
+  if need_to_run
```

Comment:
@underscorgan I think it would make these tests easier to read if this were a guard clause rather than needing to nest the rest of the test inside this if-statement. Something like the [`pass_test` method](https://github.com/puppetlabs/beaker/blob/master/lib/beaker/dsl/outcomes.rb#L51) should get you what you'd need:
```ruby
pass_test(msg) unless need_to_run
```
Other than that note, this all looks good to me 👍

Reply:
oh, cool, I didn't know that existed. Thanks @kevpl !